go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -860,6 +861,46 @@ func (root *RootValue) UpdateSuperSchemasFromOther(ctx context.Context, tblNames
 	return newRootValue(root.vrw, newRootSt), nil
 }
 
+func (root *RootValue) RenameTable(ctx context.Context, oldName, newName string) (*RootValue, error) {
```

Comment:
Add a method doc comment about why this is necessary (instead of RemoveTable, PutTable)